pkg/sink/cloudevent/cloudevent.go

```diff
@@ -0,0 +1,73 @@
+/*
+Copyright 2021 The Tekton Authors
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package cloudevent
+
+import (
+	"context"
+	"fmt"
+	"time"
+
+	cloudevents "github.com/cloudevents/sdk-go/v2"
+	"go.uber.org/zap"
+
+	triggersv1 "github.com/tektoncd/triggers/pkg/apis/triggers/v1beta1"
+)
+
+const (
+	// TriggerProcessingStarted is sent for Sink triggers received which
+	// are being processed
+	TriggerProcessingStartedV1 = "triggers.tekton.eventlistener.triggerprocessing.started.v1"
+	// TaskRunSuccessfulEventV1 is sent for Sink Triggers when a trigger is succssful
+	TriggerProcessingSuccessfulV1 = "triggers.tektoneventlistener.triggerprocessing.successful.v1"
+	// TriggerProcessingFailedEventV1 is sent for Sink Triggers when we fail to process trigger
+	TriggerProcessingFailedEventV1 = "triggers.tekton.eventlistener.triggerprocessing.failed.v1"
+)
+
+// CEClient matches the `Client` interface from github.com/cloudevents/sdk-go/v2/cloudevents
+type CEClient cloudevents.Client
+
+// SendCloudEvents is used by the EventListener to send cloud events status of
+// trigger processing
+func SendCloudEvents(payload []byte, el triggersv1.EventListener, eventID, eventType, targetURI string, ceClient CEClient, logger *zap.SugaredLogger) {
+	logger = logger.With(zap.String("trigger", el.Name))
+
+	event := cloudevents.NewEvent()
+	event.SetID(eventID)
+	event.SetSubject(el.Name + " processing " + eventID)
```

Comment:
The subject in Pipeline CEs is just the run name. Should it just be the eventID here? (thoughts @afrittoli ?)